src/bblgum/core.clj

```diff
@@ -1,8 +1,8 @@
 (ns bblgum.core
   (:require [bblgum.impl :as i]))
 
-(defn gum
-  "Main driver fn for using gum: https://github.com/charmbracelet/gum
+(defn- gum*
```

Comment:
we can merge this and the `gum` right? move this logic to the end of the multi-arity dispatch and remove `gum*`?

Reply:
We call the `gum*` function from each arity separately, so it makes sense to keep it this way. And since it's not public, it doesn't matter much, does it?

Reply:
Although give me a minute

Reply:
well the thought process i have is since this is non breaking, no one really minds this little indirection and no one else is gonna call it so inline it.

Reply:
i quite like the `impl` style ns than private fns, quite the good separation and can test more things.

Reply:
Ok, I got rid of multiple arities and gum* in one go.